numpy/core/numeric.py

```diff
@@ -862,7 +862,7 @@ def flatnonzero(a):
     array([-2, -1,  1,  2])
 
     """
-    return a.ravel().nonzero()[0]
+    return asanyarray(a).ravel().nonzero()[0]
```

Comment:
Using `np.ravel(a)` here instead will fix the bug I mentioned above

Reply:
One way to satisfy me and avoid the try/except is to write this using more elementary functions: `np.nonzero(np.ravel(a))[0]`

Reply:
Which is better here: `np.ravel(a).nonzero()[0]` or `np.nonzero(np.ravel(a))[0]`?

Reply:
It doesn't really make any difference right now, but I suppose the latter is more future-proof?